MiniApp/Classes/core/Storage/MiniAppKeyChain.swift

```diff
@@ -74,11 +76,78 @@ import Foundation
         return supportedPermissionList
     }
 
-    private func write(keys: KeysDictionary) {
+    private func writePermissionInfo(keys: KeysDictionary) {
         guard let data = try? JSONEncoder().encode(keys) else {
             return
         }
+        setValueInKeyChain(data: data)
+    }
+
+    private func retrieveAllPermissions() -> KeysDictionary? {
+        guard let storedData = getAllKeysFromKeychain() else {
+            return nil
+        }
+
+        guard let keys = ResponseDecoder.decode(decodeType: KeysDictionary.self, data: storedData) else {
+            return nil
+        }
+
+        return keys
+    }
+
+    // MARK: - Cache Verifier Methods
+    internal func setCacheInfo(key: String, for keyId: String) {
+        var keysDic = getAllCacheKeys()
```

Comment:
Could you add unit tests for this functionality? The RSDKUtils repo has unit tests, so maybe you can copy some of those?

Reply:
Working..